### PR TITLE
Bugfix/GTC-1000: 

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

When resuming after a previous aborted run, pixetl does NOT include the previously generated tiles in tiles.geojson

Issue Number: N/A

## What is the new behavior?

- When resuming after a previous aborted run, pixetl DOES include the previously generated tiles in tiles.geojson. It does this by adding a new classification of tile status, "existing".

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: tests/test_pipe.py [hunk 1]
```diff
@@ -98,18 +100,27 @@ def test_filter_target_tiles():
...
+            i += 1
+            assert isinstance(tile, Tile)
+    assert i == 4
+
 
 def test_upload_file():
     tiles = _get_subset_tiles()
-    with mock.patch.object(Tile, "upload", return_value=None):
-
+    # FIXME: I can't get this mock to work, despite trying to patch in many places
```
Comment (thomas-maschler) [inline tests/test_pipe.py:114]: is comment still valid?

File: tests/test_raster_pipe.py [hunk 2]
```diff
@@ -48,14 +48,13 @@ def test_create_tiles_subset():
...
-                skipped_tiles,
-                failed_tiles,
-            ) = PIPE.create_tiles(overwrite=True)
+            (tiles, skipped_tiles, failed_tiles, existing_tiles) = PIPE.create_tiles(
+                overwrite=True
+            )
             assert len(tiles) == 1
             assert len(skipped_tiles) == 3
             assert len(failed_tiles) == 0
+            assert len(existing_tiles) == 0
```
Comment (thomas-maschler) [inline tests/test_raster_pipe.py:57]: this or a similar test would be a good place to test if existing tiles are correctly identified and carries through the entire pipe.

File: tests/test_upload_geometries.py [hunk 3]
```diff
@@ -11,27 +12,29 @@
...
+                                [-20.0, 0.0],
+                                [-20.0, 10.0],
+                            ]
+                        ],
+                    },
+                    "properties": {"name": "/vsis3/gfw-data-lake-test/10N_020W.tif"},
+                }
+            ],
+        }
+    )
```
Comment (thomas-maschler) [inline tests/test_upload_geometries.py:37]: Do we still need this test? I believe we substituted it by fetching the metadata early on.

Comment (thomas-maschler) [reply tests/test_upload_geometries.py:37]: There should be a test instead which checks that existing tiles and current tiles are correctly merged into one geojson

**Review by thomas-maschler — CHANGES_REQUESTED** [review]

File: tests/test_pipe.py (hunk 1 above)
Comment (dmannarino) [reply tests/test_pipe.py:114]: Afraid so. It's a pre-existing test I tried modifying to both improve and extend to cover existing tiles. But the mock as written doesn't work, and I can't figure out why.

File: tests/test_upload_geometries.py (hunk 3 above)
Comment (dmannarino) [reply tests/test_upload_geometries.py]: Yup. Done.

File: tests/test_raster_pipe.py (hunk 2 above)
Comment (dmannarino) [reply tests/test_raster_pipe.py:65]: Done.

**Comment by thomas-maschler** [comment]
Something got mess up and conflicts still show although already resolved. Will reopen another PR using the same branch

**Pull request closed without merging**